thrust/thrust/system/cuda/detail/dispatch.h

```diff
@@ -26,96 +26,186 @@
 #  pragma system_header
 #endif // no system header
 
+#include <thrust/detail/integer_math.h>
 #include <thrust/detail/integer_traits.h>
 #include <thrust/detail/preprocessor.h>
 
+#include <cuda/std/detail/libcxx/include/stdexcept>
```

Comment:
Is this `detail` header something you want to replace with a public header, per https://github.com/NVIDIA/cccl/pull/2310#discussion_r1736273655?

Reply:
There is no public header for that right now